authentication.md

```diff
@@ -138,6 +138,12 @@ There are multiple ways of transmitting access tokens:
 * as an `application/x-www-form-urlencoded` parameter: `access_token=[access token]`
 * inside the request header: `Authorization: Bearer [access token]`
 
+## Authorization errors
+
+Most of the API nodes require authentication, and if the token wasn't provided or it was wrong the server will reply with status code `401 Unauthorized`.
```

Comment:
All API nodes require authentication. If no token or an invlaid token is supplied, the API will always return `401 Unauthorized`.

Reply:
How about requesting public content, e.g. public posts? Isn't it possible without authentication? In VK.com API it was possible to request public content without authorization.

Reply:
I'm on the fence here. Generally requiring authentication makes implementing rate limiting and the like easier should the need arise. Also we could argue that dynamic client registration makes it easier to get started with anything. But then that somewhat nullifies the former argument and you still have to fallback to using the IP/subnet, too at least. 

I guess currently this means even requesting public content has to be done on the behalf of a particular user, which doesn't immediatly sound right. 

Anyway, we can see this is a more complex topic and we shouldn't discuss this in this pull request. Let's stick with Dennis comment here.

Reply:
I actually didn't think too much about this, it was just a gut feeling. But I tend towards agreeing with @jhass :) But yeah, this should be something for Discourse. I feel like... trying requiring auth, and if a use case comes up, talk about endpoints that don't require auth? :)